spring-cloud-aws-s3-parent/spring-cloud-aws-s3/src/main/java/io/awspring/cloud/s3/S3Operations.java

```diff
@@ -60,7 +61,17 @@ public interface S3Operations {
 	 * @param s3Url - the S3 url s3://bucket/key
 	 */
 	void deleteObject(String s3Url);
-
+	
+	/**
+	 * Returns some or all (up to 1,000) of the objects in a bucket.
+	 * Does not handle pagination. If you need pagination you should use {@link S3PathMatchingResourcePatternResolver} or {@link S3Client}
```

Comment:
Maybe we should include a paginator request since it will paginate if there are more than 1000 objects internally ->
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/s3/S3Client.html#listObjectsV2Paginator(software.amazon.awssdk.services.s3.model.ListObjectsV2Request)

Reply:
According to the issue I used as a reference: https://github.com/awspring/spring-cloud-aws/issues/767 . It is specified that in the `S3Template`, pagination would not be managed on the list objects. If you need pagination, you should use the `S3Client` or `S3PathMatchingResourcePatternResolver` according to the issue.
I can manage the pagination if required. Thanks.